projects/ngrx-auto-entity/src/lib/service.ts

```diff
@@ -58,21 +59,41 @@ export interface IAutoEntityService<TModel> {
 
   loadRange?(entityInfo: IEntityInfo, range: Range, criteria?: any): Observable<IEntityWithRangeInfo<TModel>>;
 
-  create?(entityInfo: IEntityInfo, entity: TModel, criteria?: any): Observable<TModel>;
+  create?(entityInfo: IEntityInfo, entity: TModel, criteria?: any, originalEntity?: TModel): Observable<TModel>;
```

Comment:
Should there be 2 distinct types here for `entity` and `originalEntity`? I.E. The entity before transformation might be of type `interface TModelBeforeTransform { date: string }` whereas the transformed entity would be `interface TModelAfterTransform { date: Date }`.

Reply:
Technically speaking, there probably should be. However within the framework itself, I could not actually find a way to thread that type into the transformation process. Mainly because the transformations are attached to the model via the decorator, and the model is technically not defined yet when the decorator is being configured.

I guess what I should probably do...is type the entity now as `TModel | any`. That would maintain backwards compatibility, but then also allow arbitrary data, and thus hopefully let the end developer actually type this however they please if they need to.